src/index.ts

```diff
@@ -19,58 +20,78 @@ async function mergePdfBuffers(pdfBuffers: Array<Buffer>) {
   }
 
   return outDoc.save();
-};
+}
 
 const getURL = (origin: string, filePath: string) => {
-  return origin + "/" + filePath.substring(filePath.startsWith('/') ? 1 : 0);
-}
+  return origin + "/" + filePath.substring(filePath.startsWith("/") ? 1 : 0);
+};
 
 const getStylesheetPathFromHTML = (html: string, origin: string) => {
-  let regExp = /(?:|<link.*){1}href="(.*styles.*?\.css){1}"/g;
+  const regExp = /(?:|<link.*){1}href="(.*styles.*?\.css){1}"/g;
   let filePath = "";
   try {
     filePath = getFirstCapturingGroup(regExp, html);
   } catch {
-    throw new Error("The href attribute of the 'styles*.css' file could not be found!");
+    throw new Error(
+      "The href attribute of the 'styles*.css' file could not be found!"
+    );
   }
   return getURL(origin, filePath);
 };
 
 const getScriptPathFromHTML = (html: string, origin: string) => {
-  let regExp = /(?:|<script.*){1}src="(.*styles.*?\.js){1}"/g;
+  const regExp = /(?:|<script.*){1}src="(.*styles.*?\.js){1}"/g;
   let filePath = "";
   try {
     filePath = getFirstCapturingGroup(regExp, html);
   } catch {
-    throw new Error("The src attribute of the 'styles*.js' file could not be found!");
+    throw new Error(
+      "The src attribute of the 'styles*.js' file could not be found!"
+    );
   }
   return getURL(origin, filePath);
 };
 
 const getFirstCapturingGroup = (regExp: RegExp, text: string) => {
-  let match = regExp.exec(text);
+  const match = regExp.exec(text);
   if (match && match[1]) {
     return match[1];
   } else {
-    throw new ReferenceError("No capture group found in the provided text.")
+    throw new ReferenceError("No capture group found in the provided text.");
   }
+};
+
+function isObject(x: unknown): x is Record<PropertyKey, unknown> {
+  return x !== null && typeof x === "object";
 }
 
-const isAddressInfo = (arg: any): arg is AddressInfo => {
-  return arg
-    && arg.address && typeof (arg.address) == 'string'
-    && arg.family && typeof (arg.family) == 'string'
-    && arg.port && typeof (arg.port) == 'number';
+function hasOwnProperty<
+  X extends Record<PropertyKey, unknown>,
+  Y extends PropertyKey
+>(obj: X, prop: Y): obj is X & Record<Y, unknown> {
+  return Object.prototype.hasOwnProperty.call(obj, prop);
 }
 
-export function getPathSegment(path: string, endSlash: boolean = true) {
+const isAddressInfo = (arg: unknown): arg is AddressInfo => {
+  return (
+    isObject(arg) &&
+    hasOwnProperty(arg, "address") &&
+    typeof arg.address == "string" &&
+    hasOwnProperty(arg, "family") &&
+    typeof arg.family == "string" &&
+    hasOwnProperty(arg, "port") &&
+    typeof arg.port == "number"
+  );
+};
```

Comment:
The only big thing I changed was the `isAddressInfo = (arg: any)` to `(arg: unknown)`, since eslint didn't like me to use `any`, and some fixes the make the TypeScript compiler happy.